### PR TITLE
Add coverage to unit tests

### PR DESCRIPTION
**Pull request by desislavaa**
**Description**
The new delivery program expects coverage reports for Compass components.

Changes proposed in this pull request:
- Lightweight code coverage implementation. It was decided that we'll stick to this approach until more sophisticated code coverage reporting is required.

image

### CODE REVIEW

File: scripts/generic_make_go.mk [hunk 1]
```diff
@@ -175,9 +175,16 @@ endif
...
 $(foreach t,$(COPY_TARGETS),$(eval $(call buildpack-cp-ro,$(t))))
 
-test-local:
+test-local-no-coverage:
 	go test ./...
 
+test-local:
+	@go test ./... -coverprofile cover.out.tmp
+	@cat cover.out.tmp | grep -v "_gen.go" | grep -v "hack" > cover.out
+	@echo -n Code Coverage:
```
Comment (la4ezar) [inline scripts/generic_make_go.mk:184]: The output will look better if it is 
@echo -n 'Code Coverage: '

File: scripts/generic_make_go.mk [hunk 2]
```diff
@@ -175,9 +175,16 @@ endif
...
-test-local:
+test-local-no-coverage:
 	go test ./...
 
+test-local:
+	@go test ./... -coverprofile cover.out.tmp
+	@cat cover.out.tmp | grep -v "_gen.go" | grep -v "hack" > cover.out
+	@echo -n Code Coverage: 
+	@go tool cover -func cover.out | grep total | tr -s "[:blank:]" | tr "[:blank:]" " " | cut -d " " -f 3
+	@rm cover.out.tmp cover.out
```
Comment (la4ezar) [inline scripts/generic_make_go.mk:186]: If 'go test ...' fails this will not be executed and the 'cover.out.tmp' file will not be deleted.

**Review by la4ezar — CHANGES_REQUESTED** [review]

File: scripts/generic_make_go.mk (hunk 2 above)
Comment (la4ezar) [reply scripts/generic_make_go.mk:186]: You can change the 'go test...' with `@go test ./... -coverprofile cover.out.tmp || true`

Comment (DimitarPetrov) [reply scripts/generic_make_go.mk:186]: This will prevent prow jobs from failing when tests are failing.

Comment (desislavaa) [reply scripts/generic_make_go.mk:186]: for prow jobs - it doesn't matter, for local dev - I've added it into gitignore - is that ok?

**Review by la4ezar — APPROVED** [review]

**Pull request merged**